testcases/open_posix_testsuite/conformance/interfaces/strchr/1-1.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * This file is licensed under the GPL license. For the full content
+ * of this license, see the COPYING file at the top level of this
+ * source tree.
+ */
+
+/*
+ * assertion:
+ *
+ *	The function will locate the first occurrence of c (converted to a char)
+ *	in the string pointed to by s. The terminating NUL character is
+ *	considered to be part of the string. Upon completion, the function will
+ *	return a pointer to the byte, or a null pointer if the byte was not found.
+ *
+ * method:
+ *	-Generate sample string s1
+ *	-Define one char, taking a character from the sample string(char1), one char as NULL(char2)
+	and third character out of the sample string(char3).
+ *	-Use strchr for string with char1 and store result in result1.
+ *	-Use strchr for string with char2 and store result in result2.
+ *	-Use strchr for string with char3 and store result in result3.
+ *	-Compare the result1 with pointer to the defined char(char1).
+ *	-Compare the result2 with pointer to the defined char(char2).
+ *	-Compare result3 with NULL.
+ *	-Repeat the above all steps for given number of iterations.
+
+*/
+
+#define _XOPEN_SOURCE 700
```

Comment:
Not sure, which standard you want to test, but for POSIX 2008 (`_XOPEN_SOURCE 700`) it complain about redefinition. IMHO it should be `_XOPEN_SOURCE     600` (like the rest of the code) or omit it.

Reply:
@anitha-suresh FYI: abac60a2e openposix: Remove _XOPEN_SOURCE in strchr/strcpy/strncpy
=> IMHO flags need to be changed in build system or undef _XOPEN_SOURCE before redefining it in the source.